pricegraph/src/lib.rs

```diff
@@ -134,6 +134,59 @@ impl Pricegraph {
             sell: sell_amount,
         })
     }
+
+    /// Computes a transitive orderbook for the given market specified by the
+    /// base and quote tokens.
+    ///
+    /// This method optionally accepts a spread is a decimal fraction that
+    /// defines the maximume transitive order price with the equation:
+    /// `first_transitive_price + first_transitive_price * spread`. This means
+    /// that given a spread of 0.5 (or 50%), and if the cheapest transitive
+    /// order has a price of 1.2, then the maximum price will be `1.8`.
+    ///
+    /// The spread applies to both `asks` and `bids` transitive orders.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if the spread is zero or negative.
+    pub fn transitive_orderbook(
+        &self,
+        base: TokenId,
+        quote: TokenId,
+        spread: Option<f64>,
+    ) -> TransitiveOrderbook {
+        let mut orderbook = self.full_orderbook();
+
+        let mut transitive_orderbook =
+            orderbook.reduce_overlapping_transitive_orderbook(base, quote);
+        transitive_orderbook
+            .asks
+            .extend(orderbook.clone().fill_transitive_orders(
+                TokenPair {
+                    buy: base,
+                    sell: quote,
+                },
+                spread,
+            ));
+        transitive_orderbook
+            .bids
+            .extend(orderbook.fill_transitive_orders(
+                TokenPair {
+                    buy: quote,
+                    sell: base,
+                },
+                spread,
+            ));
+
+        for orders in &mut [
+            &mut transitive_orderbook.asks,
+            &mut transitive_orderbook.bids,
+        ] {
+            orders.sort_unstable_by(|a, b| num::compare(a.price(), b.price()));
```

Comment:
If I understand correctly, the prices for bids and asks are going to be represented as inverse prices (thus the value will jump drastically between best bid/best ask). 
I believe normally bids and asks in an orderbook have the same price unit (e.g. https://poloniex.com/public?command=returnOrderBook&currencyPair=BTC_ETH&depth=50)

Reply:
One thing to note is that we return ask and bid orders, not prices. That is buy and sell amounts (as they are represented in the exchange).

I guess, one thing we can add is `inverse` method for inverting the orders and calculating the inverse price.

The sorting here is just so the ask orders are sorted, and the bid orders are sorted, and they are sorted interdependently.

Reply:
Maybe we can change the `TransitiveOrderbook` representation to align with expectations a bit more? We can also add iterator methods to produce `volume/price` pairs for the price estimator service?

Reply:
Created #1002  to capture this work.